manifests/init.pp

```diff
@@ -376,6 +376,10 @@
     if $agent_major_version != undef and $agent_major_version != $_agent_major_version {
       fail('Provided and deduced agent_major_version don\'t match')
     }
+    if $facts['os']['name'] == 'CentOS' and $facts['os']['name']['distro']['release']['major'] < 7 and 0 + $3 > 51 {
```

Comment:
Shouldn't we be covering more OS's of the same family here? These at least, probably: https://github.com/DataDog/puppet-datadog-agent/blob/5b111de56b0706a76d6fbfba218040953b487280/manifests/redhat.pp#L38

Reply:
It also looks like we're missing the version capping when agent_version is actually undefined or set to latest. I.e. the equivalent to [this](https://github.com/DataDog/chef-datadog/blob/4221b8c563ca11d3f152d8b8d61eb99e13263dfa/recipes/_install-linux.rb#L56-L59).